network/check.py

```diff
@@ -325,68 +325,61 @@ def _check_linux(self, instance):
                 }
                 self._submit_devicemetrics(iface, metrics)
 
-        try:
-            proc_snmp_path = "{}/net/snmp".format(proc_location)
-            proc = open(proc_snmp_path, 'r')
-
-            # IP:      Forwarding   DefaultTTL InReceives     InHdrErrors  ...
-            # IP:      2            64         377145470      0            ...
-            # Icmp:    InMsgs       InErrors   InDestUnreachs InTimeExcds  ...
-            # Icmp:    1644495      1238       1643257        0            ...
-            # IcmpMsg: InType3      OutType3
-            # IcmpMsg: 1643257      1643257
-            # Tcp:     RtoAlgorithm RtoMin     RtoMax         MaxConn      ...
-            # Tcp:     1            200        120000         -1           ...
-            # Udp:     InDatagrams  NoPorts    InErrors       OutDatagrams ...
-            # Udp:     24249494     1643257    0              25892947     ...
-            # UdpLite: InDatagrams  Noports    InErrors       OutDatagrams ...
-            # UdpLite: 0            0          0              0            ...
-            try:
-                lines = proc.readlines()
-            finally:
-                proc.close()
 
-            tcp_lines = [line for line in lines if line.startswith('Tcp:')]
-            udp_lines = [line for line in lines if line.startswith('Udp:')]
 
-            tcp_column_names = tcp_lines[0].strip().split()
-            tcp_values = tcp_lines[1].strip().split()
-            tcp_metrics = dict(zip(tcp_column_names, tcp_values))
-
-            udp_column_names = udp_lines[0].strip().split()
-            udp_values = udp_lines[1].strip().split()
-            udp_metrics = dict(zip(udp_column_names, udp_values))
+        netstat_data = {}
+        for f in ['netstat', 'snmp']:
+            proc_data_path = "{}/net/{}".format(proc_location, f)
+            netstat = open(proc_data_path, 'r')
```

Comment:
Can you do this inside the try/catch? The `except IOError` probably will never trigger otherwise.